vaadin-text-field-flow-parent/vaadin-text-field-flow/src/main/java/com/vaadin/flow/component/textfield/TextFieldValidationSupport.java

```diff
@@ -77,14 +77,14 @@ ValidationResult checkValidity(String value) {
             return ValidationResult.error("");
         }
 
-        final boolean isMinLengthNotReached = value != null && minLength != null
-                && value.length() < minLength;
+        final boolean isMinLengthNotReached = value != null && !value.isEmpty()
```

Comment:
Nit: Could extract `value != null && !value.isEmpty()` into a variable with a readable name like `hasValue`, and then reuse below.

Reply:
If I do so, it starts complaining about potential null pointer access:

<img width="687" alt="image" src="https://user-images.githubusercontent.com/5039436/205004406-57a03837-7b5d-420f-b6e5-f61247649bdf.png">

Do you know how it could be worked around?

Reply:
Perhaps using Optional could help.

Reply:
Let's probably keep it the way it is